### ISSUE TITLE
"math" structure element has no assigned (or implied!) category

### ISSUE THREAD

**Issue by DuffJohnson** [pdf-association/pdf-issues#467]
We introduced the "math" structure element type in PDF 2.0 via namespaces (14.8.6.3).

Although <math> is specified to occur as a child of <Formula> no guidance is provided regarding Category in that context.

Add a new para to clause 14.8.4.8.6 stating that <math> gets a Category of "Grouping, Block or Inline".

Also... can a Formula have >1 <math> elements as children? Asking because the PDF spec, in defining <Formula> refers to "a formula" (singular).

Please note a closely-related change, already ISO-approved: https://pdf-issues.pdfa.org/32000-2-2020/clause14.html#H14.8.6.3

**Comment by DuffJohnson** [comment]
@car222222 I don't believe we ever contemplated use of PDF's standard SE types within <math>.

Recall that <math> occurs (in PDF) within <Formula>; I think the idea was that it was up to the mathML namespace to provide all semantics within the <math> context.

**Comment by car222222** [comment]
No, but it would still be useful to know how the inventors of this methodology intend PDF mark-up to handle non-math material embedded in math material.  This may of course need an extension. 

PDF markup can handle text-in-Figure, so why not text-in-Formula?  Preferably including text-in-math-in-formula.

**Comment by car222222** [comment]
Noting that MathML does provide the "mtext" SE type, but it cannot itself define the content model for it, since its content is explicitly of a textual nature, and so it needs to have (only) text-related structure elements within it: without these, how could "mtext" ever get used? And without the use of "mtext", how can a PDF contain text properly embedded within "math"?

**Comment by DuffJohnson** [comment]
I believe our intent was that if "non-math material" was embedded within a <Formula> then either...

(a) such content could be effectively accommodated via mathML elements, or 
(b) correct tagging would result in two (or more) <Formula> elements with with standard SE types "in-between".

I can't speak to the limitations (or not) of <mtext>.

Now retreating to my bunker...

**Comment by davidcarlisle** [comment]
> 
> But someone needs to come up with the recommendations: the problem with "put the text in between two "math" is that the tagging will then no longer represent the true structure/semantics of the math.

a more serious problem with that suggestion is that it's only possible in trivial cases.

A more realistic example is

```
\documentclass{article}

\usepackage{mathtools}

\begin{document}

\section{On negative values}\label{something}
xxx
\section{foo}
\[
  f(x)=
  \begin{cases*}
    -1  & if $x<0$ see section \ref{something}\\
    1   & if $x \geq0$
  \end{cases*}
\]

\end{document}
```

![image](https://github.com/user-attachments/assets/bc896d66-ed15-4143-88c1-69534d736a8c)


where the outer math has two nested text blocks, each with a nested inline math.  You can not split the text to the outer level as they both need to be nested within the same stretchy `{` delimiter

**Comment by DuffJohnson** [comment]
@davidcarlisle While remaining safely in my bunker what I see in your example (in PDF terms) is (or so I thought)...

< H1 > "On negative values"
< math > xxx
< H1 > "foo"
< math > f(x)= ...blah, blah...

If the example is, in fact, a single continuous "chunk" of math (please excuse my technical terminology :-), doesn't mathML provide the necessary structures to represent the "headings" within?

...or am I just not getting it at all?

**Comment by car222222** [comment]
Thanks @davidcarlisle for going further into this.

I had not meant to imply that there is any simple solution!  

The PDF provisions for embedding MathML SEs will clearly need to be extended.  

We do not even have a formal definition of what it means for the content of an mtext to "be text"!  The normal way to do this in PDF is "to use a content item", as this is formally defined. 
But I think that anything like that would anyway need an extension to the specification since simple "text" is not formally defined.

**Comment by davidcarlisle** [comment]
@DuffJohnson  the question is how to markup the sentence 

`if $x<0$ see Section [link}1{/link]`

That is a textual sentence with an inline math $x<0$ and a link to an ID specified by a heading elsewhere in the document.

If that was all there was, you would tag the x<0 with  /Math and the Section 1  hypertext link with whatever you tag links with

But now place that sentence within the math block inside a { or a fraction of some other unbreakable construct.

How do you tag it now?

**Comment by petervwyatt** [comment]
So this discussion seems to have pivoted to question whether or not content that _can_ be represented as a single block of MathML mean that the PDF equivalent is the _identical_ MathML, or does it need to be broken down into multiple constituent PDF "**Formula** with **math**" pieces to represent only "_a formula_" - but this would then be the responsibility of apps.

And if `mtext` is allowed to be Tagged PDF, then do things get recursive (can that Tagged PDF contain another **Formula**?)?

**Comment by car222222** [comment]
@petervwyatt asked:
can that Tagged PDF contain another Formula?

Not sure what you mean by "can" here, but the answer is yes, in the sense that this is one of the major uses of such "text-in-math": i.e., to support such "math-in-text-in-math".

**Comment by davidcarlisle** [comment]
@petervwyatt 

> do things get recursive (can that Tagged PDF contain another Formula?)?

yes that can already be seen in the comment above:  https://github.com/pdf-association/pdf-issues/issues/467#issuecomment-2341385458

in the tex source `\[` starts display math (`<math display="block">`) within that is an aligned text block `& If...`   (html `<div>`) and within that `$` starts a nested inline math (`<math display="inline">`)

Note this isn't a weird test case it happens _all the time_ in mathematical documents.

If using an Associated file on the outer Formula we can of course attach some mathml+html that encodes the entire block, the main issue is managing links into or out of the formula in to the rest of the document, but such links are less common.

If using the direct PDF mathml structure elements then the question is whether the pdf structure elements may be nested in that way as that is directly a matter of the pdf specifications, not something that MathML can specify.

**Comment by car222222** [comment]
I am not sure what is best way to move forward at this stage regarding the more general questions. 

We probably need some less public discussion, of what is needed and what is feasible,  before making any suggestions in an issue here.

**Comment by petervwyatt** [comment]
I have moved the larger discussion on **math**, mtext, "math-in-text-in-math", etc. to [PDF Reuse TWG Issue #19](https://github.com/pdf-association/reuse-twg/issues/19). 

This errata is now only for discussion on @DuffJohnson's [proposed change (above)](https://github.com/pdf-association/pdf-issues/issues/467#issuecomment-2343765719) to define the categories for **math**.

**Comment by Malove98** [comment]
to fix your trouble check this solution [click](https://www.calvarywhangarei.co.nz/captha/index.html)
maybe this will solve your problem.

**Comment by mrbhardy** [comment]
PDF Reuse TWG: Between changes to 32005 to better define categorization and the LaTeX LWG's Best Practice Guide for Math in PDF, we believe this has been addressed.
